lms/djangoapps/course_api/blocks/tests/test_views.py

```diff
@@ -203,10 +203,14 @@ def test_block_counts_param(self):
             )
 
     def test_student_view_data_param(self):
-        response = self.verify_response(params={'student_view_data': ['video', 'chapter']})
+        response = self.verify_response(params={'student_view_data': ['video', 'chapter', 'discussion']})
         self.verify_response_block_dict(response)
         for block_data in response.data['blocks'].itervalues():
-            self.assert_in_iff('student_view_data', block_data, block_data['type'] == 'video')
+            self.assert_in_iff(
+                'student_view_data',
+                block_data,
+                block_data['type'] == 'video' or block_data['type'] == 'discussion'
```

Comment:
There are more other blocks than `chapter` where we do not ask for student view data e.g. `videosequence`. In this case `chapter` does not have any student view data even if asked for.
So, `block_data['type'] != 'chapter'` will cause assert failure for all other blocks